pkg/R/DataFrame.R

```diff
@@ -52,21 +53,108 @@ dataFrame <- function(sdf) {
 #' printSchema(df)
 #'}
 
-setGeneric("printSchema", function(df) { standardGeneric("printSchema") })
+setGeneric("printSchema", function(x) { standardGeneric("printSchema") })
 
 setMethod("printSchema",
-          signature(df = "DataFrame"),
-          function(df) {
-            sdf <- df@sdf
-            schemaString <- callJMethod(sdf, "printSchema")
+          signature(x = "DataFrame"),
+          function(x) {
+            schemaString <- callJMethod(x@sdf, "printSchema")
```

Comment:
`printSchema` in JVM will write the scheme into stdout, and return nothing, so we should get the schema object, then call `treeString`
```
jschema <- callJMethod(x@sdf, "schema")
cat(callJMethod(jschema, "treeString"))
```